app/services/discovery_report.rb

```diff
@@ -6,6 +6,7 @@
 #   report.to_builder.target!  # generates the report as a JSON string
 class DiscoveryReport
   attr_reader :batch, :start_time, :summary
+  attr_accessor :had_errors, :error_message
 
```

Comment:
same theory as above, we will track errors and messages for discovery reports too